### DISCUSSION TITLE
Unable to Find Windows 64-bit Package for pyARTS 2.6.6 – Need Installation Guidance

### DISCUSSION THREAD

**Discussion by jayeshgit65** [atmtools/arts#815] [Q&A] [answered]
Hello everyone,

I am currently trying to install pyARTS 2.6.6 on my Windows 64-bit system, but I am facing some difficulties. While browsing through the available packages, I noticed that the files are mostly labeled for osx-arm64 and linux-64 architectures, and there doesn't seem to be a package explicitly for Windows (win-64). Following is the site link where i searched :- https://anaconda.org/rttools/pyarts/files
                           Could anyone guide me on the correct steps to install pyARTS 2.6.6 on Windows 64-bit? Is there a Windows-specific package available, or would I need to build from source? Any help or suggestions would be greatly appreciated!

Thank you in advance for your assistance!

**Comment by riclarsson** [comment, marked as answer]
Hi,

We currently do not support version 2.6 for windows. We do not plan to support windows for this version of pyarts.

You would need to not just compile from source but also fix the C++ and make to make this work.

**Comment by stefanbuehler** [comment]
Hi,

I do not recommend trying to build from source for Windows for that version. There are reasons, why we do not offer the packages.

Note that version three will run on windows, we are already in the process of testing this. So, if you are experienced and adventurous, it would make more sense to try with the development version than with 2.6.

Stefan

**Reply by jayeshgit65** [reply]
Hello,

Thank you for the information regarding the availability of pyARTS 2.6.6 for Windows. As it's clear that the package isn't available for Windows 64-bit. I want to ask you about one thing that the installation process of pyARTS 2.6.6 is quite puzzling. Despite having a Windows 64-bit system, I noticed that I was able to complete the installation using the macOS-specific package. However, issues arose when I attempted to use those installations for further working.
Could you please clarify why this package begins to install on Windows ?
Thank you for your help!
![Screenshot 2024-09-07 162857](https://github.com/user-attachments/assets/c11004c8-7588-47dc-a393-3877b2306a81)
![WhatsApp Image 2024-09-07 at 19 28 12_2e6a87ed](https://github.com/user-attachments/assets/143c6838-9b43-486f-8e36-98f69877a2ed)

**Reply by olemke** [reply]
This is a design decision by the conda developers. When you download a package instead of installing it from a repository, conda allows you to install any package without further compatibility checks. Also be aware that installing the package this way will not install any required dependencies. You're basically circumventing a lot of features conda otherwise provides.

**Reply by jayeshgit65** [reply]
thank you sir for your valuable answer! 
sir can you please guide me in installation of pyarts 2.6.6 from source for my redhat linux 64 bit system .I am trying to install it ,but number of errors are continuosly arising . I look forward for your support through which i can install it. 

Thank you for your help!

**Reply by riclarsson** [reply]
@JQYESH422x , during which point of the [installation guide](https://github.com/atmtools/arts/tree/v2.6.6?tab=readme-ov-file#building-arts) did it fail?  What was the error message?

**Reply by jayeshgit65** [reply]
As I am working under government organisations. Our devices are not connected to internet . We usually download all the files we need from one windows setup which has internet connection and transfer it to our Linux system . During installation ,  I am getting error on command "mamba env update -f environment-dev-linux.yml" environment-dev-linux.yml this file consist of dependency but it's giving error that https error occured as it is not connected to the internet . Can you please ellaborate from where do i get all such dependencies to download and share to my offline setup of redhat 8 linux 64bit system .  when i tried to run cd arts ,followed by the cmake --preset=default -gcc -conda also giving error that source directory not found (i have allready provided absolute path to art2.6x) .

**Reply by riclarsson** [reply]
I am unable to help you more.

We provide a list of dependencies.  These and all of their dependencies (recursively) must be available to run and build ARTS.  How you ensure that in an artificially limited environment is completely up to you.
